app/parse.py

```diff
@@ -8,8 +24,121 @@ class Quote:
     tags: list[str]
 
 
+@dataclass
+class Author:
+    name: str
+    born: str
+    description: str
+
+
+def parse_single_quote(quote_soup: Tag) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.get_text() for tag in quote_soup.select(".tag")],
+    )
+
+
+def prettify_description(text: str) -> str:
+    """Mainly to remove unexpected whitespace characters"""
+    return " ".join(text.split()).replace("''", '"')
+
+
+def parse_single_author(author_href: str) -> Author:
+    page = requests.get(BASE_URL + author_href).content
+    author_page_soup = BeautifulSoup(page, "html.parser")
+
+    author = Author(
+        name=author_page_soup.select_one(".author-title").text,
+        born=author_page_soup.select_one(".author-born-date").text
+        + " "
+        + author_page_soup.select_one(".author-born-location").text,
+        description=prettify_description(
+            author_page_soup.select_one(".author-description").text
+        ),
+    )
+
+    logging.info(f"Adding author: {author}")
+
+    return author
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> [Quote]:
+    quotes = page_soup.select(".quote")
+
+    return [parse_single_quote(quote) for quote in quotes]
+
+
+def get_single_page_authors(
+    page_soup: BeautifulSoup, cached_author_hrefs: [str] = None
+) -> [Author]:
```

Comment:
Better to create `QuoteScraper`  class and use OOP principles,  than just writing functions